ioapic: no id sanity check

82marbag opened the pull request:
*Issue #, if available:* #175

*Description of changes:*

Remove the double check on a device's IOAPIC ID as it breaks
on systems where the ID returned by a device differs from
the one read at boot from the MADT tables

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

wipawel commented on arch/x86/ioapic.c:
```diff
@@ -176,12 +176,7 @@ void __text_init init_ioapic(void) {
         ioapic_id_t id;
 
         id.reg = ioapic_read32(ioapic, IOAPIC_ID);
-        if (ioapic->id != id.apic_id) {
-            panic("IOAPIC with unexpected APIC ID detected: 0x%02x (expected: "
-                  "0x%02x)\n",
-                  id.apic_id, ioapic->id);
-        }
```
Maybe keep it as a warning message

82marbag replied:
Updated

82marbag committed:
ioapic: no id sanity check

Remove the double check on a device's IOAPIC ID as it breaks
on systems where the ID returned by a device differs from
the one read at boot from the MADT tables

Signed-off-by: Daniele Ahmed <ahmeddan amazon c;0m >
